Added WeakSet and Emmet Support for JSX

zaynali53 opened the pull request:
@zertosh Emmet's default is to not support JS files and i end up defining a keybinding provided by <a href='https://github.com/allanhortle'>Allan Hortle</a> in his <a href='https://github.com/allanhortle/JSX'>JSX</a> package. I also find the scope defined in the .tmLanguage so i added source.js.jsx there and removed keybinding. It works the same.

zertosh commented:
@zaynali53 Thanks! Though I'm not totally sold on the `scopeName` change, I'll go with it - if it causes any issues to non-emmet users, then I'll revert it

zertosh commented:
published at v6.2.0 - it should get picked up by Package Control soon

zaynali53 commented:
Sure. Thanks @zertosh

zertosh commented:
Sorry @zaynali53, I had to revert the `scopeName` change. See https://github.com/babel/babel-sublime/issues/124

zaynali53 commented:
Yeah got it @zertosh 
Worth mentioning here <a target='_blank' href='https://github.com/sergeche/emmet-sublime/blob/master/README.md#how-to-expand-abbreviations-with-tab-in-other-syntaxes'>How to expand abbreviations with Tab in other syntaxes</a>.
I updated my emmet package, changed my keybinding and it's working well with source.js scope. 
```js
{
    "keys": ["tab"],
    "command": "expand_abbreviation_by_tab",
    "context": [
        {
            "operand": "source.js",
            "operator": "equal",
            "match_all": true,
            "key": "selector"
        }
    ]
}
```
You can add this keybinding for JSX syntax support since JS isn't supported in Emmet.
